### ISSUE TITLE
Suggestions for combining {DrWatson} with static site generator for a kind virtual lab book

### ISSUE THREAD

**Issue by wetlandscapes** [JuliaDynamics/DrWatson.jl#340]
Hi,
Thank you very much for the excellent package!

I was wondering if you all have suggestions for integrating `{DrWatson}` with a static site generator. Specifically, I was thinking of using notebook-based analysis (.jmd files) that could be converted to static web pages to better interlace code, narrative, exploratory plots, etc. with things like function and type documentation. Maybe also include journaling and literature review sections to keep track of ideas and what I've been doing. Effectively I'm looking to make a kind of virtual lab book in an attempt to further open up my science.

I'm still relatively new to Julia, so don't know all the options that are out there, but the first thing I came across was `{Franklin}`, which seems like it could be combined with `{DrWatson}` for the kind of site I envision. But I'm wondering if you all might know of some other options that are out there. For example, maybe `{Documentor}` would be a better option, since it is built specifically for the function and type documenting I mention? Or maybe there is a way to hybridize all the above?

If this isn't the kind of thing you care to comment on, no worries! I can close the issue. Just thought I'd see if you had a perspective on the idea.

**Comment by Datseris** [comment]
Hello, and thank you for the kind words!

I do not believe that there is anything additional required from the side of DrWatson. As you suggested, Franklin is a nice tool, but I think Documenter.jl + Literate.jl is even better, already works with the function-based design of DrWatson, and in addition allows straightforwardly hosting documentation via Documenter.jl and GitHub. In fact, the documentation of DrWatson itself is made this way. If you look at the documentation of DrWatson, does it satisfy your needs? Especially the workflow tutorial page, is nothing more than a standard and runnable Julia file, that is expanded into a "static site" via Literate+Documenter:  https://github.com/JuliaDynamics/DrWatson.jl/blob/master/docs/src/workflow.jl

**Comment by wetlandscapes** [comment]
Thanks for pointing me in the Documenter + Literate workflow. I've not yet had a reason to get very deep into Literate, so this is the kick I may have needed.

I also recently came across https://quarto.org/, which also looks interesting. At this point, my projects are multilingual due to different collaborators having different language competencies. So being able to run notebooks using native interpreters, rather than through interop functionalities, may be the more friendly option (though Quarto seems much slower than any of the Julia stuff I've used so far).

So, thank you for the insights!
